hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/FusiformSimilarityTraverser.java

```diff
@@ -0,0 +1,144 @@
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.tinkerpop.gremlin.structure.Edge;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+import org.apache.tinkerpop.gremlin.util.iterator.IteratorUtils;
+
+import com.baidu.hugegraph.HugeGraph;
+import com.baidu.hugegraph.backend.id.Id;
+import com.baidu.hugegraph.structure.HugeEdge;
+import com.baidu.hugegraph.structure.HugeVertex;
+import com.baidu.hugegraph.type.define.Directions;
+import com.baidu.hugegraph.util.CollectionUtil;
+import com.baidu.hugegraph.util.E;
+
+public class FusiformSimilarityTraverser extends HugeTraverser {
+
+    public FusiformSimilarityTraverser(HugeGraph graph) {
+        super(graph);
+    }
+
+    public Map<Id, Set<Id>> fusiformSimilarity(List<HugeVertex> vertices,
+                                               Directions direction, Id label,
+                                               int minEdgeCount, long degree,
+                                               float alpha, int top,
+                                               String groupProperty,
+                                               int minGroupCount,
+                                               long capacity, long limit) {
+        checkCapacity(capacity);
+        checkLimit(limit);
+        int loop = 0;
+        long accessed = vertices.size();
+        Map<Id, Set<Id>> results = new LinkedHashMap<>(vertices.size());
+        for (HugeVertex vertex : vertices) {
+            Iterator<Edge> edges = this.edgesOfVertex(vertex.id(), direction,
+                                                      label, minEdgeCount);
+            long edgeCount = IteratorUtils.count(edges);
+            if (edgeCount < minEdgeCount) {
+                // Ignore current vertex if its matched edges number not enough
+                continue;
+            }
+
+            // Get similar nodes and counts
+            edges = this.edgesOfVertex(vertex.id(), direction, label, degree);
+            Map<Id, Integer> similars = new HashMap<>();
+            int neighborCount = 0;
+            while (edges.hasNext()) {
+                neighborCount++;
+                checkCapacity(capacity, ++accessed, "fusiform similarity");
+                Id target = ((HugeEdge) edges.next()).id().otherVertexId();
+                Directions backDir = direction.opposite();
+                Iterator<Edge> backEdges = this.edgesOfVertex(target, backDir,
+                                                              label, degree);
+                while (backEdges.hasNext()) {
+                    HugeEdge backEdge = (HugeEdge) backEdges.next();
+                    Id node = backEdge.id().otherVertexId();
+                    if (similars.containsKey(node)) {
+                        similars.put(node, similars.get(node) + 1);
```

Comment:
add class MutableInteger with increase() method, to avoid put the counts again:
```
MutableInteger counts = similars.get(node)
if (counts == null) {
    similars.put(node, new MutableInteger(0))
    checkCapacity()
}
counts.increase()
```